components/ai_chat/core/browser/engine/oai_api_client.cc

```diff
@@ -122,6 +125,22 @@ void OAIAPIClient::PerformRequest(
   }
 }
 
+std::optional<std::string> OAIAPIClient::ExtractErrorMessage(
+    const std::string& result) {
+  auto value = base::JSONReader::Read(result);
+
+  if (!value.has_value() && !value->is_dict()) {
+    return std::nullopt;
+  }
+
+  auto* error = value->GetDict().FindDict("error");
+  if (error) {
+    auto* message = error->FindString("message");
+    return *message;
```

Comment:
message could be nullptr in theory, we shouldn't dereference it directly without checking.

Reply:
These error cases should be covered in a unit test.

Reply:
added a check